daemons/based/based_remote.c

```diff
@@ -333,10 +343,15 @@ cib_remote_listen(gpointer data)
 void
 cib_remote_connection_destroy(gpointer user_data)
 {
-    pcmk__client_t *client = user_data;
+    struct new_client_data_s *ncd = user_data;
+    pcmk__client_t *client = ncd->new_client;
     int csock = 0;
 
     if (client == NULL) {
+        if (ncd != NULL) {
+            free(ncd);
+        }
```

Comment:
free() can handle NULL, no need to check (and that's what coverity is complaining about, ncd was already dereferenced above)